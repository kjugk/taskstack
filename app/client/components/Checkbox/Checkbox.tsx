import React from 'react';
import Icon from 'semantic-ui-react/dist/commonjs/elements/Icon';
import { withTheme } from 'styled-components';

interface Props {
  checked: boolean;
  theme: any;
  onClick(): any;
}

const Checkbox: React.SFC<Props> = ({ checked, onClick, theme }) => (
  <a
    href="#"
    onClick={(e) => {
      e.preventDefault();
      e.stopPropagation();
      onClick();
    }}
  >
    {checked && <Icon name="check square" style={{ color: theme.main }} />}
    {!checked && <Icon name="square outline" style={{ color: theme.black }} />}
  </a>
);

export default withTheme(Checkbox);
