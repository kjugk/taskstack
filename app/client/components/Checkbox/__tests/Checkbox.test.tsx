import * as React from 'react';
import { shallow, mount } from 'enzyme';
import Checkbox from '../Checkbox';

const onClick = jest.fn();

describe('<Checkbox />', () => {
  describe('renders correctly', () => {
    const wrapper = shallow(<Checkbox checked={false} onClick={onClick} theme={{}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('fire onClick', () => {
    const wrapper = mount(<Checkbox checked={true} onClick={onClick} theme={{}} />);
    wrapper.find('a').simulate('click');
    expect(onClick).toBeCalled();
  });

  describe('icon', () => {
    // Icon が切り替わるか
    it('have checkbox icon', () => {
      const wrapper = mount(<Checkbox checked={true} onClick={onClick} theme={{}} />);
      expect(wrapper.find('i.check')).toHaveLength(1);
    });

    it('have square icon', () => {
      const wrapper = mount(<Checkbox checked={false} onClick={onClick} theme={{}} />);
      expect(wrapper.find('i.square.outline')).toHaveLength(1);
    });
  });
});
