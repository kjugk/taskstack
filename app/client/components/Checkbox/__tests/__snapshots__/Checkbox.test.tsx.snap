// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Checkbox
  checked={false}
  onClick={[MockFunction]}
  theme={Object {}}
/>
`;
